crates/forge_main/src/main.rs

```diff
@@ -10,8 +10,22 @@ use forge_main::{Cli, Sandbox, TitleDisplayExt, UI, tracker};
 
 #[tokio::main]
 async fn main() -> Result<()> {
+    // Set up signal handler for graceful cursor restoration on Ctrl+C
+    tokio::spawn(async {
+        if tokio::signal::ctrl_c().await.is_ok() {
+            // Ensure cursor is visible before exiting
+            use crossterm::{cursor::Show, execute};
+            let _ = execute!(std::io::stdout(), Show);
+            std::process::exit(130); // Standard exit code for SIGINT
+        }
+    });
+
     // Set up panic hook for better error display
     panic::set_hook(Box::new(|panic_info| {
+        // Ensure cursor is visible before panicking
+        use crossterm::{cursor::Show, execute};
+        let _ = execute!(std::io::stdout(), Show);
+
```

Comment:
I think we should handle this in the UI widget that's actually causing this issue, instead of a global hook.